docs/modules/qbi-helper/interest-template.md

```diff
@@ -0,0 +1,146 @@
+* pass the component ID as the first argument
+
+For example, to add a query you might do one of the following:
+
+```csharp
+// Component identified by type argument
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Component identified by ID given as first argument
+InterestTemplate.Create()
+    .AddQueries(Position.ComponentId, query1, query2);
+```
+
+### Add
+
+When adding queries, you can either provide the queries as parameters or an enumerable set:
+
+```csharp
+// Parameters
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Enumerable
+var queryList = new List() { query1, query2 };
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(queryList);
+```
+
+### Replace
+
+This operation replaces all the existing queries for the given component ID with the new queries you pass in. In a similar way to adding queries, you can provide the queries as parameters or an enumerable set.
+
+```csharp
+// Parameters
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(query1);
+
+// Enumerable
+var queryList = new List() { query1 };
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(queryList);
+```
+
+> At least one query must be provided, otherwise the existing queries for a given authoritative component will not be removed.
+
+### Clear
+
+You can either clear queries for a given authoritative component or clear _all_ queries in the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+```csharp
+// Removes all queries added for the Position component
+basicInterestTemplate.ClearQueries<Position.Component>();
+
+// Removes all queries
+basicInterestTemplate.ClearAllQueries();
+```
+
+## Get Interest from the template
+
+After adding, removing or modifying a set of queries, there are two ways to get `Interest` out of the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+### Snapshot
+
+To return a [`Snapshot`]({{urlRoot}}/api/core/snapshot) of the `Interest` component that can be used when defining entity templates, call `ToSnapshot()` on the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+```csharp
+// Create an interest template
+var interestTemplate = InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Add the Interest component to an entity template
+var entityTemplate = new EntityTemplate();
+...
+entityTemplate.AddComponent(interestTemplate.ToSnapshot(), WorkerUtils.UnityGameLogic);
+```
+
+### ComponentInterest
+
+To update `Interest` at execution time, you need to modify the underlying Dictionary within the `Interest` component.
+
+The [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) class provides the `AsComponentInterest` method to return this Dictionary, so that it can be used to update the `Interest` component.
```

Comment:
```suggestion
The [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) class provides the `AsComponentInterest` method to return this Dictionary. This can be used to update the `Interest` component.
```